### PR TITLE
(docs) Added component templates to contribution guidance

### PR DESCRIPTION
**Pull request by connor-leech**
Added component templates to contribution guidance.

### CODE REVIEW

File: src/pages/how-to-contribute/component/index.mdx [hunk 1]
```diff
@@ -19,19 +25,622 @@ To contribute a component to Carbon, start by [opening a Github issue](https://g
 - Include mockups of any fidelity (optional)
 - Include any inspirations from other products (optional)
 
-This issue will be the staging ground for the contribution and an opportunity for the community to weigh in with any suggestions. We'll also be able to tell you up-front whether the contribution is likely to become part of Carbon, and potentially save you time.
+This issue will be the staging ground for the contribution and an opportunity for the community to weigh in with any suggestions. We encourage you to surface work in progress. Someone in the community may be working on the same component and interested in collaborating with you.
+
+## Documentation templates
+
+Our component usage documentation typically contains three parts: usage guidance, style guidance, and code guidance. Some components are more complex than others but you should try to cover each topic included in these templates.
```
Comment (janchild) [inline src/pages/how-to-contribute/component/index.mdx:32]:
```suggestion
Our component usage documentation typically contains three parts: usage guidance, style guidance, and code guidance. Some components are more complex than others but you should cover each topic included in these templates.
```

File: src/pages/how-to-contribute/component/index.mdx [hunk 2]
```diff
@@ -19,19 +25,622 @@ To contribute a component to Carbon, start by [opening a Github issue](https://g
...
+
+### Usage guidance for components with one type
+
+Use this template for documenting components with only one type. You can see this in use for [Accordion](/components/accordion) and [Checkbox](/components/checkbox).
```
Comment (janchild) [inline src/pages/how-to-contribute/component/index.mdx:36]:
```suggestion
Use this template for documenting components with only one type, that is, no variants. You can see this template in use for [Accordion](/components/accordion) and [Checkbox](/components/checkbox).
```

File: src/pages/how-to-contribute/component/index.mdx [hunk 3]
```diff
@@ -19,19 +25,622 @@ To contribute a component to Carbon, start by [opening a Github issue](https://g
...
+## Feedback
+
+<!--Include a request for feedback here.-->
+
+Help us improve this component by providing feedback, asking questions, and leaving any other comments on [GitHub](https://github.com/carbon-design-system/carbon-website/issues/new?assignees=&labels=feedback&template=feedback.md).
+```
+
+### Usage guidance with multiple types
+
+Use this template for documenting components that have multiple types. You can see this in use for [Dropdown](/components/dropdown/usage), [Modal](/components/modal/usage), and [Notification](/components/notification/usage).
```
Comment (janchild) [inline src/pages/how-to-contribute/component/index.mdx:270]:
```suggestion
Use this template for documenting components that have multiple types or variants. You can see this template in use for [Dropdown](/components/dropdown/usage), [Modal](/components/modal/usage), and [Notification](/components/notification/usage).
```

File: src/pages/how-to-contribute/component/index.mdx [hunk 4]
```diff
@@ -19,19 +25,622 @@ To contribute a component to Carbon, start by [opening a Github issue](https://g
...
+## Feedback
+
+Help us improve this component by providing feedback, asking questions,
+and leaving any other comments on [GitHub](https://github.com/carbon-design-system/carbon-website/issues/new?assignees=&labels=feedback&template=feedback.md).
+```
 
 ## Parts of a component contribution
 
-Here are all of the steps for a complete contribution.
+Component contributions would ideally include all of the following pieces/parts.
```
Comment (janchild) [inline src/pages/how-to-contribute/component/index.mdx:637]:
```suggestion
Component contributions ideally include all of the following parts.
```

File: src/pages/how-to-contribute/component/index.mdx [hunk 5]
```diff
@@ -97,3 +704,7 @@ The component or enhancement must be built in one of our supported frameworks (V
   </ResourceCard>
 </Column>
 </Row>
+
+#### 6. Provide code documentation
```
Comment (janchild) [inline src/pages/how-to-contribute/component/index.mdx:708]:
```suggestion
#### 6. Code documentation
```

File: src/pages/how-to-contribute/component/index.mdx [hunk 6]
```diff
@@ -19,19 +25,622 @@ To contribute a component to Carbon, start by [opening a Github issue](https://g
...
+-->
+
+## Feedback
+
+Help us improve this component by providing feedback, asking questions, and leaving any other comments on [GitHub](https://github.com/carbon-design-system/carbon-website/issues/new?assignees=&labels=feedback&template=feedback.md).
+```
+
+### Style guidance
+
+We're still working on this template. We'd love your help and feedback!
```
Comment (janchild) [inline src/pages/how-to-contribute/component/index.mdx:530]: @connor-leech do we want to add a call to action here?

Comment (connor-leech) [reply src/pages/how-to-contribute/component/index.mdx]: @janchild Just added a link to an issue 👍

**Pull request merged**